waku/v2/node/storage/sqlite.nim

```diff
@@ -35,6 +35,9 @@ template dispose(db: Sqlite) =
 template dispose(db: RawStmtPtr) =
   discard sqlite3_finalize(db)
 
+template dispose*(db: SqliteStmt) =
```

Comment:
On a sidenote (not part of this PR but also regarding template usage): there is

```nim
template dispose(db: Sqlite) =
  discard sqlite3_close(db)

```

and (see added comment)

```nim
proc close*(db: SqliteDatabase) =
  discard sqlite3_close(db.env)    # <- this could use the template above

  db[] = SqliteDatabase()[]

```

Reply:
> Just a nitpick: Is there a reason the parameter of this template (and the one above) are called db? It seems, they are statements.

Good point. Copied pattern from above, but both should indeed be called something other than `db` :). [4261f66](https://github.com/status-im/nwaku/pull/981/commits/4261f66f7c6a9695751f25e8f6ca900a592cd5b2)

> Also, there are some direct calls of discard sqlite3_finalize(...) on sqlite3 statements (e.g. in line 91),

I am OK with these direct `sqlite3_` calls _internally_ - for me the main purpose of the exported `dispose` is to have a more convenient way to finalize prepared statements (and discard the result) from other modules (to follow a similar pattern [to nimbus](https://github.com/status-im/nimbus-eth2/blob/f124f22f108f64416f890297560cae8e6c136d08/beacon_chain/beacon_chain_db.nim#L274-L276)). Agree with you though that we should be consistent, so have replaced the internal use of `dispose` with direct calls in [4261f66](https://github.com/status-im/nwaku/pull/981/commits/4261f66f7c6a9695751f25e8f6ca900a592cd5b2).

There is still the matter of using the internal `dispose` to close the DB, but this can indeed be addressed in subsequent results.